Engine/ac/route_finder_impl.cpp

```diff
@@ -0,0 +1,270 @@
+//=============================================================================
+//
+// Adventure Game Studio (AGS)
+//
+// Copyright (C) 1999-2011 Chris Jones and 2011-20xx others
+// The full list of copyright holders can be found in the Copyright.txt
+// file, which is part of this source code distribution.
+//
+// The AGS source code is provided under the Artistic License 2.0.
+// A copy of this license can be found in the file License.txt and at
+// http://www.opensource.org/licenses/artistic-license-2.0.php
+//
+//=============================================================================
+//
+// New jump point search (JPS) A* pathfinder by Martin Sedlak.
+//
+//=============================================================================
+
+#include "ac/route_finder_impl.h"
+
+#include <string.h>
+#include <math.h>
+
+#include "ac/common.h"   // quit()
+#include "ac/movelist.h"     // MoveList
+#include "ac/common_defines.h"
+#include "gfx/bitmap.h"
+
+#include "route_finder_jps.inl"
+
+extern MoveList *mls;
+
+#ifdef DEBUG_PATHFINDER
+extern void Display(const char *, ...);
+#endif
+
+using AGS::Common::Bitmap;
+
+namespace AGS {
+namespace Engine {
+namespace RouteFinder {
+
+#define MAKE_INTCOORD(x,y) (((unsigned short)x << 16) | ((unsigned short)y))
+
+static const int MAXNAVPOINTS = MAXNEEDSTAGES;
+static int navpoints[MAXNAVPOINTS];
+static int num_navpoints;
+static fixed move_speed_x, move_speed_y;
+static Navigation nav;
+static AGS::Common::Bitmap *wallscreen;
```

Comment:
Please note that you have `using AGS::Common::Bitmap;` above, so you don't have to repeat namespaces here and below. Perhaps same could be done for other ***.cpp** files too.